core/lib/zksync_core/src/l1_gas_price/gas_adjuster/mod.rs

```diff
@@ -133,6 +166,55 @@ impl GasAdjuster {
         // For now, pubdata is only sent via calldata, so its price is pegged to the L1 gas price.
         self.estimate_effective_gas_price() * L1_GAS_PER_PUBDATA_BYTE as u64
     }
+
+    async fn get_blob_base_fee_history(
+        eth_client: &Arc<dyn EthInterface>,
+        block_range: Range<usize>,
+    ) -> Result<Vec<U256>, Error> {
+        let mut history = Vec::new();
+        for block_number in block_range {
+            let header = eth_client
+                .block(U64::from(block_number).into(), "gas_adjuster")
+                .await?;
+
+            if let Some(excess_blob_gas) = header.and_then(|header| header.excess_blob_gas) {
+                history.push(Self::blob_base_fee(excess_blob_gas.as_u64()))
+            }
+        }
+
+        Ok(history)
+    }
+
+    /// Calculates `blob_base_fee` given `excess_blob_gas`.
+    fn blob_base_fee(excess_blob_gas: u64) -> U256 {
```

Comment:
let's add a test case for this - so that we know that we do the correct computations